### PR TITLE
Catalog of my things

### PR DESCRIPTION
**Pull request by zhorabay**
- [ ] In this project, we:
- Inserted and queried data in SQL.
- Used primary key & foreign key mechanisms for joining tables.
- Queried multiple tables.
- Run a program using the command line.
- Used Ruby syntax for basic programming operations.
- Applied Ruby best practices and language style guides in code.
- Described the SOLID principles of OOP.
- Implemented classes and objects in Ruby.
- Implemented encapsulation and inheritance with Ruby.
- Set up associations between classes and objects.
- Stored data in .json files.
- Built interactive console apps.

### CODE REVIEW

File: classes/music_album.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  attr_reader :id, :on_spotify
+
+  def initialize(publish_date, on_spotify)
+    super(publish_date)
+    @on_spotify = on_spotify
+  end
+
+  def can_be_archived?
+    super == true && @on_spotify == true
+  end
```
Comment (Gambit142) [inline classes/music_album.rb:14]: - [ ] Please note that the `can_be_archived?` method should be a private method based on the schema diagram provided. Please implement this change. 👍 😎

File: classes/item.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @author = author
+    author.items << self unless author.items.include?(self)
+  end
+
+  def label=(label)
+    @label = label
+    label.items << self unless label.items.include?(self)
+  end
+
+  def move_to_archive
```
Comment (Gambit142) [inline classes/item.rb:28]: - [ ] Kindly write tests for all the attributes and public methods of the `Item` class. Remember that you ought to write tests for all classes in this project. Kindly implement this. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @zhorabay @Chimoneg27 @jubaer919 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Great use of GitHub flow ✔️ 
- Created necessary classes, attributes and methods ✔️ 
- Created a console that functions properly ✔️ 
- Created SQL tables that mirror classes and their attributes ✔️ 
- Wrote tests using Rspec testing library ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: classes/item.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    @archived = true if can_be_archived?
+  end
+
+  private
+
+  def can_be_archived?
+    difference = Time.now.year - @publish_date.year
+    difference >= 10
+  end
```
[the remark this answers is not in the document]
Comment (Abenezer-Tilahun) [reply classes/item.rb]: Fixed.

File: classes/game.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+  def initialize(publish_date, multiplayer, last_played_at)
+    super(publish_date)
+    @multiplayer = multiplayer
+    @last_played_at = last_played_at
+  end
+
+  def aligable
+    (Date.today.year - @last_played_at.year) > 2
+  end
```
[the remark this answers is not in the document]
Comment (Abenezer-Tilahun) [reply classes/game.rb]: Fixed.

File: README.md [hunk 5]
```diff
@@ -1 +1,185 @@
...
+```
+
+### Usage
+
+- To run the project, execute the following command:
+
+```sh
+ cd [YOUR FOLDER]
+ git clone https://github.com/zhorabay/Catalog.git
+```
```
[the remark this answers is not in the document]
Comment (Abenezer-Tilahun) [reply README.md]: Fixed.

File: schema.sql [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  last_name VARCHAR(100) NOT NULL
+);
+
+CREATE TABLE game (
+  id INT NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  publish_date DATE NOT NULL,
+  multiplayer VARCHAR(100) NOT NULL,
+  last_played_at DATE NOT NULL,
+  FOREIGN KEY (id) REFERENCES items(id)
+);
```
[the remark this answers is not in the document]
Comment (Abenezer-Tilahun) [reply schema.sql]: Fixed

File: schema.sql [hunk 7]
```diff
@@ -0,0 +1,52 @@
+CREATE TABLE items (
+  id INT NOT NULL GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+  genre_id INT,
+  author_id INT,
+  author VARCHAR(50) NOT NULL,
+  source VARCHAR(50) NOT NULL,
+  label VARCHAR(50) NOT NULL,
```
[the remark this answers is not in the document]
Comment (Abenezer-Tilahun) [reply schema.sql]: Fixed.

File: classes/music_album.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply classes/music_album.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @zhorabay @jubaer919 @Chimoneg27 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.
✔️ All Test Cases are passing.
image



## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

**Pull request merged**